drivers/led_strip/led_strip_shell.c

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2025 Dmitrii Sharshakov <d3dx12.xx@gmail.com>
+ *
+ * Based on drivers/led/led_shell.c which is:
+ * Copyright (c) 2020 Seagate Technology LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "zephyr/kernel.h"
+#include "zephyr/sys/util.h"
+#include <zephyr/shell/shell.h>
+#include <zephyr/drivers/led_strip.h>
+#include <stdlib.h>
+
+#define LOG_LEVEL CONFIG_LOG_DEFAULT_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(led_shell);
+
+#define MAX_COLOR_ARGS 255
+
+enum {
+	arg_idx_dev	= 1,
+	arg_idx_value	= 2,
+};
+
+static int parse_common_args(const struct shell *sh, char **argv,
+			     const struct device * *dev)
+{
+	*dev = shell_device_get_binding(argv[arg_idx_dev]);
+	if (!*dev) {
+		shell_error(sh,
+			    "LED device %s not found", argv[arg_idx_dev]);
+		return -ENODEV;
+	}
+
+	return 0;
+}
+
+static int cmd_update_rgb(const struct shell *sh, size_t argc, char **argv)
+{
+	const struct device *dev;
+	int err;
+	size_t num_colors, strip_len;
+	uint8_t i;
+	struct led_rgb *pixels;
+
+	err = parse_common_args(sh, argv, &dev);
+	if (err < 0) {
+		return err;
+	}
+
+	strip_len = led_strip_length(dev);
+	num_colors = argc - arg_idx_value;
+	if (num_colors > MIN(strip_len, MAX_COLOR_ARGS)) {
+		shell_error(sh,
+			"Invalid number of colors %d (max %d)",
+			num_colors, MIN(strip_len, MAX_COLOR_ARGS));
+		return -EINVAL;
+	}
+
+	pixels = k_malloc(num_colors * sizeof(struct led_rgb));
+
+	for (i = 0; i < num_colors; i++) {
+		if (strlen(argv[arg_idx_value + i]) < 6) {
+			shell_error(sh, "Invalid color length for value %s, expected 6",
+				argv[arg_idx_value + i]);
+			return -EINVAL;
+		}
+
+		err = hex2bin(argv[arg_idx_value + i], 2, &pixels[i].r, 1);
+		if (err == 0) {
+			shell_error(sh, "Invalid red channel parameter %s",
+				argv[arg_idx_value + i]);
+			return -EINVAL;
+		}
+
+		err = hex2bin(argv[arg_idx_value + i] + 2, 2, &pixels[i].g, 1);
+		if (err == 0) {
+			shell_error(sh, "Invalid green channel parameter %s",
+				argv[arg_idx_value + i]);
+			return -EINVAL;
+		}
+
+		err = hex2bin(argv[arg_idx_value + i] + 4, 2, &pixels[i].b, 1);
+		if (err == 0) {
+			shell_error(sh, "Invalid blue channel parameter %s",
+				argv[arg_idx_value + i]);
+			return -EINVAL;
+		}
```

Comment:
This is quite some code duplication, an alternative could be reading all 6 digits into a single uint32_t and then using bytes from it as r, g and b. Please let me know what would be preferred